bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -5937,6 +6047,52 @@ private static void closeStatement(Statement statement) {
 		}
 	}
 
+	void setIngestJobEndDate(int ingestJobId, long endDate) throws TskCoreException, TskDataException {
+		CaseDbConnection connection = connections.getConnection();
+		acquireSharedLock();
+		ResultSet resultSet = null;
+		try {
+			Statement statement = connection.createStatement();
+			resultSet = statement.executeQuery("SELECT ingest_job_id FROM ingest_jobs WHERE ingest_job_id=" + ingestJobId);
+			if (!resultSet.next()) {
+				if (resultSet.getInt(1) == 0) {
+					statement.executeUpdate("UPDATE ingest_jobs SET end_date=" + endDate + " WHERE ingest_job_id=" + ingestJobId + ";");
+				} else {
+					throw new TskDataException("Given ingest job has already been finished.");
+				}
+			} else {
+				throw new TskDataException("Given ingest job was not found in database.");
+			}
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error updating the end date.", ex);
+		} finally {
+			closeResultSet(resultSet);
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
+	void setIngestStatus(int ingestJobId, IngestStatusType status) throws TskCoreException, TskDataException {
+		CaseDbConnection connection = connections.getConnection();
```

Comment:
We can eliminate the TskDataException throw and the SELECT. The method is not public and the constructed object that calls it should know its id, which is set by the SleuthkitCase instance via construcotr and immutable..